server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -296,4 +299,27 @@ private ImmutableList<AccountModel> searchAccounts(
                                 searchParameters.nameQuery().get().toLowerCase(Locale.ROOT)))))
         .collect(ImmutableList.toImmutableList());
   }
+
+  private boolean checkIfEmptySearchParams(SearchParameters searchParameters) {
```

Comment:
I think this handles all the cases for returning a full list including partial dates. If yes, why should we check again during Account filtering?

Reply:
`checkIfEmptySearchParams` checks if any single one of the date fields is empty/null whereas in `searchAccounts` it checks all the date fields to make sure none of them are empty/null.  I could remove the empty string checks (not the null, though) from `searchAccounts` but, in that case, if any date field had an empty string, the user would see the error message "Please enter date in MM/dd/yyyy format", similar to them making any other formatting mistake.  

Come to think of it, that might be a good way to address the missing date field error message that you mention below.  I'd need to change the error message wording, but I think I should do that anyway.  I also really don't like that the current error message shows up on a blank page like this:
![image](https://github.com/civiform/civiform/assets/134556723/4106f08a-b651-46b5-8c51-5d6ec6524d3c)

I'll ask Sijia about that too.

Reply:
I agree!

Reply:
I consolidated the empty/blank/null checks into this method: `TrustedIntermediaryService.findMissingSearchParams` so that I am only doing those checks once.  It returns a list of missing parameters which I can then use to know specifically which parameters are missing (Name, Day, Month, or Year).